src/Persistence/Models/CustomProperty.cs

```diff
@@ -28,19 +31,37 @@ public record CustomProperty
 
     public string? Tooltip { get; init; }
 
-    public IList<CustomPropertyParameter> Parameters { get; init; } = new List<CustomPropertyParameter>();
+    public CustomPropertyParametersCollection Parameters { get; init; } = new();
 
     public enum PropertyDirection
     {
         Input,
         Output
     }
 
+    public enum PropertyCategory
```

Comment:
why not use https://github.com/microsoft/PowerApps-Tooling/blob/master/src/Persistence/Models/PropertyCategory.cs?

Reply:
I removed that, not sure why it kept showing up :)